aws_lambda_script/main.py

```diff
@@ -0,0 +1,189 @@
+SOURCE_KEY = os.environ.get("SOURCE_KEY", "new_project_data.json")
+DESTINATION_BUCKET = os.environ.get("DESTINATION_BUCKET", "sdp-dev-tech-radar")
+DESTINATION_KEY = os.environ.get("DESTINATION_KEY", "onsTechDataAdoption.csv")
+
+
+def get_json_from_s3(bucket: str, key: str) -> Dict:
+    """Retrieve JSON data from S3 bucket"""
+    try:
+        s3_client = boto3.client("s3")
+        response = s3_client.get_object(Bucket=bucket, Key=key)
+        json_data = json.loads(response["Body"].read().decode("utf-8"))
+        return json_data
+    except Exception as e:
+        logger.error("Error retrieving JSON from S3: %s", str(e))
+        raise
+
+
+def process_project_data(project_data: Dict) -> List[Dict]:
+    """Process project data into the required format"""
+    processed_data = []
+
+    projects = project_data.get("projects", [])
+    logger.info("Processing %d projects from JSON data", len(projects))
+
+    for project in projects:
+        row = {
+            "Project": project.get("details", [{}])[0].get("name", ""),
+            "Project_Short": project.get("details", [{}])[0].get("short_name", ""),
+            "Project_Area": "",
+            "DST_Area": "",
+            "Team": "",
+            "Language_Main": ";".join(
+                project.get("architecture", {}).get("languages", {}).get("main", [])
+            ),
+            "Language_Others": ";".join(
+                project.get("architecture", {}).get("languages", {}).get("others", [])
+            ),
+            "Language_Frameworks": ";".join(
+                project.get("architecture", {}).get("frameworks", {}).get("others", [])
+            ),
+            "Testing_Frameworks": "",
+            "Hosted": ";".join(
+                project.get("architecture", {}).get("hosting", {}).get("details", [])
+            ),
+            "Messaging_Type": "",
+            "Containers": "",
+            "Architectures": "",
+            "Source_Control": project.get("source_control", [{}])[0].get("type", ""),
+            "Branching_Strategy": "",
+            "Repo": project.get("source_control", [{}])[0]
+            .get("links", [{}])[0]
+            .get("url", ""),
+            "Static_Analysis": "",
+            "Code_Formatter": "",
+            "Package_Manager": "",
+            "Security_Tools": "",
+            "CICD": ";".join(
+                project.get("architecture", {}).get("cicd", {}).get("others", [])
+            ),
+            "CICD_Orchestration": "",
+            "Monitoring": "",
+            "Datastores": ";".join(
+                project.get("architecture", {}).get("database", {}).get("others", [])
+            ),
+        }
+        processed_data.append(row)
+
+    logger.info("Processed %d projects", len(processed_data))
+    return processed_data
+
+
+def write_to_s3_csv(data: List[Dict], bucket: str, key: str):
+    """Write processed data to S3 as CSV"""
+    try:
+        s3_client = boto3.client("s3")
```

Comment:
Example where global scope would be useful

Reply:
Done 👍